core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -0,0 +1,239 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of NumberWithUnits and Units
+ * domain objects.
+ */
+
+oppia.constant('NUMBER_WITH_UNITS_PARSING_ERRORS', {
+  INVALID_VALUE:
+    'Please ensure that value is either a fraction or a number',
+  INVALID_CURRENCY:
+    'Please enter a valid currency (e.g., $ 5)',
+  INVALID_UNIT_CHARS:
+    'Please ensure that unit only contains numbers, alphabets, (, ), *, ^, /, -'
+});
+
+oppia.factory('NumberWithUnitsObjectFactory', [
+  'UnitsObjectFactory', 'FractionObjectFactory',
+  'NUMBER_WITH_UNITS_PARSING_ERRORS', function(
+      UnitsObjectFactory, FractionObjectFactory,
+      NUMBER_WITH_UNITS_PARSING_ERRORS) {
+    var NumberWithUnits = function(type, real, fraction, units) {
+      this.type = type;
+      this.real = real;
+      this.fraction = fraction;
+      this.units = units;
+    };
+
+    NumberWithUnits.prototype.toString = function() {
+      var numberWithUnitsString = '';
+      if (this.type === 'real') {
+        if (this.units.trim() === '$' || this.units.trim() === 'Rs') {
```

Comment:
`units` is not a string. It is also an object.

Reply:
Ah... I see. In the object itself unit is stored as string. So we can use `trim()` for that also.

Reply:
You've got this wrong. Look my comment below.

Reply:
Done, I've changed it to Units object.